koap/src/jsMain/resources/index.html

```diff
@@ -0,0 +1,327 @@
+                            <option value="PUT">PUT</option>
+                        </select>
+                    </div>
+                    <div id="coderawdiv">
+                        <label for="coderawclass" class="vertlabel">Raw class range 0..7</label>
+                        <input type="number" min="0" max="7" id="coderawclass" class="vertinput">
+                        <label for="coderawdetail" class="vertlabel">Raw detail range 0..31</label>
+                        <input type="number" min="0" max="31" id="coderawdetail" class="vertinput">
+                    </div>
+                </td>
+            </tr>
+            <tr>
+                <th>ID</th>
+                <td><input id="msgid" type="text"></td>
+            </tr>
+            <tr>
+                <th>Token</th>
+                <td><input id="token" type="text"></td>
+            </tr>
+            <tr>
+                <th>Options</th>
+                <td>
+                    <select id="optionsselect">
+                        <option value="IfMatch">IfMatch</option>
+                        <option value="UriHost">UriHost</option>
+                        <option value="ETag">ETag</option>
+                        <option value="IfNoneMatch">IfNoneMatch</option>
+                        <option value="Observe">Observe</option>
+                        <option value="UriPort">UriPort</option>
+                        <option value="LocationPath">LocationPath</option>
+                        <option value="UriPath">UriPath</option>
+                        <option value="ContentFormat">ContentFormat</option>
+                        <option value="MaxAge">MaxAge</option>
+                        <option value="UriQuery">UriQuery</option>
+                        <option value="Accept">Accept</option>
+                        <option value="LocationQuery">LocationQuery</option>
+                        <option value="ProxyUri">ProxyUri</option>
+                        <option value="ProxyScheme">ProxyScheme</option>
+                        <option value="Size1">Size1</option>
+                    </select>
+                    <input type="text" id="optioninput">
+                </td>
+            </tr>
+            <tr>
+                <th>Payload</th>
+                <td><input id="payload" type="text"></td>
+            </tr>
+            <tr>
+                <td colspan="2">
+                    <button id="encodebutton" name="encodebutton" onclick="onEncodeClick()">Encode</button>
+                </td>
+                <td>
+                    <button id="decodeudpbutton" name="decodeudpbutton" onclick="onDecodeUdpClick()">DecodeUdp</button>
+                    <button id="decodetcpbutton" name="decodetcpbutton" onclick="onDecodeTcpClick()">DecodeTcp</button>
+                </td>
+            </tr>
+        </table>
+</div>
+<script src="koap.js"></script>
+<script type="text/javascript">
```

Comment:
Probably better to pull this out as it's own `.js` file.

Reply:
Moved inline Javascript into `webapp.js` file.